packages/cubit/lib/src/cubit.dart

```diff
@@ -47,14 +47,18 @@ abstract class Cubit<T> extends Stream<T> {
 
   /// Updates the [state] of the `cubit` to the provided [state].
   /// [emit] does nothing if the `cubit` has been closed.
+  ///
+  /// [track] can optionally be provided to determine whether the
+  /// state change should be tracked (reported to `onTransition`).
+  /// Default is `true`.
   @protected
-  void emit(T state) async {
+  void emit(T state, {bool track}) async {
```

Comment:
I'd rather default using:
```suggestion
  void emit(T state, {bool track = true}) async {
```
Currently, when calling `emit(state + 1, track: null)` this leads to the state change being tracked, which is undesirable.

Reply:
Yeah I agree. The only reason I chose to go with this approach is because bloc will be extending Cubit and it felt a bit weird to have the following

@override
void emit(State state, {bool track = true}) {
  super.emit(state, track: false);
}

I basically was trying to have cubit default to true while making it less awkward for classes which extend to cubit to avoid tracking (otherwise you’d see double the transitions).

Normally, I would have totally gone with the suggested approach but it’s a bit of a special case and I’m still brainstorming better/more elegant approaches. Thoughts?

Reply:
Ah, that clarifies the situation!

I don't really see a problem with the following:
```dart
@override
void emit(State state) => super.emit(state, track: false);
```
Or did you mean it feels weird to have to specify `track` in the override?

Reply:
That’s not a valid override of emit haha

Reply:
Whoops, copied it from the bloc codebase which of course doesn't include these `track` changes yet.

Reply:
Another option would be to make `bool track` nullable (as it currently is) and explicitly check for `track == true` in the `if` statement. Basically the only issue we had now is state changes being tracked when passing `track: null`.

Reply:
That was intentional though because I want that to be the default. Developers shouldn't need to call `emit(state, track: true)`.

Reply:
Hmm, yeah I see now.

I can't think of a more elegant approach at the moment.

Reply:
The other thing to consider is whether we want to force this notion of events within Cubit in the first place. It might just be best to accept the fact that events don't exist in cubit-land 😄

Reply:
Good point. From an architecture side of view, Cubit really doesn't need to take care of this. Plus Bloc is only one possible "user" of Cubit. We should let Bloc and other libraries to extend Cubit themselves.

Reply:
How exactly does this relate to tracking transitions? I suppose we still want to track those, right?

Reply:
I think that you could still track transitions in Cubit, but without the notion of an event. That would be something that Bloc adds in. 

Also, to accommodate the difference between Cubit and Bloc when it comes to transitions, it might be simpler to split Cubit into two things: a low level custom stream wrapper and a user facing Cubit class. Then Bloc could use the lower level stream wrapper and layer in Events and its own notion of transitions.